test/plugin/base_test.rb

```diff
@@ -605,6 +605,56 @@ def test_sanitize_tags_from_container_name_with_require_valid_tags_false
     end
   end
 
+  def test_configure_split_logs_by_tag
+    setup_gce_metadata_stubs
+    {
+      APPLICATION_DEFAULT_CONFIG => true,
+      DISABLE_SPLIT_LOGS_BY_TAG_CONFIG => false
+    }.each do |(config, split_logs_by_tag)|
+      d = create_driver(config)
+      assert_equal split_logs_by_tag,
+                   d.instance.instance_variable_get(:@split_logs_by_tag)
+    end
+  end
+
+  def test_split_logs_by_tag
+    setup_gce_metadata_stubs
```

Comment:
This should probably be within the loop... Feel free to defer until you do the `new_stub_context` cleanup.

Reply:
Will defer to a follow up PR.